corehq/apps/domain/views.py

```diff
@@ -2097,23 +2097,25 @@ def unpack_fuzzies(query_dict):
             Builds an integer-keyed dictionary from POST request data, and returns a list of dictionaries that can
             be wrapped by CaseSearchConfigJSON
             """
-            pattern = re.compile(r'^config\[fuzzy_properties]\[(?P<index>\d+)]\[(?P<attr>\w+)]$')
+            # match "config[fuzzy_properties][0][case_type]" and "config[fuzzy_properties][0][properties][]" but
+            # not "enable"
+            pattern = re.compile(r'^config\[fuzzy_properties]\[(?P<index>\d+)]\[(?P<attr>\w+)](?:\[])?$')
             fuzzy_dict = defaultdict(dict)
-            for key, value in query_dict.items():
+            for key in query_dict:
                 match = pattern.match(key)
-                if match:  # non-fuzzy-properties won't match, i.e. "^enable$"
-                    fuzzy_dict[int(match.group('index'))][match.group('attr')] = value
+                if match:
+                    i = int(match.group('index'))
+                    attr = match.group('attr')
+                    is_list = key.endswith('[]')  # i.e. "...[properties][]"
+                    fuzzy_dict[i][attr] = query_dict.getlist(key) if is_list else query_dict[key]
             if not fuzzy_dict:
                 return []
             return [fuzzy_dict[i] for i in range(max(fuzzy_dict.keys()) + 1) if fuzzy_dict[i]]
```

Comment:
I think you could achieve the same thing here more pythonically with `[item for key, item in fuzzy_dict.iter_items() if item]`, unless I've misunderstood what this is doing